src/libcollections/string.rs

```diff
@@ -741,6 +742,24 @@ impl String {
             string: self_ptr,
         }
     }
+
+    /// Converts the string into `Box<str>`.
+    ///
+    /// Note that this will drop any excess capacity.
+    #[unstable(feature = "box_str",
+               reason = "recently added, matches RFC")]
+    pub fn into_boxed_slice(mut self) -> Box<str> {
+        self.vec.shrink_to_fit();
+
+        let slice: Box<str> = unsafe {
+            let raw = &mut *self.vec as *mut [u8] as *mut str;
+            Box::from_raw(raw)
+        };
+
+        mem::forget(self);
+
+        slice
```

Comment:
Like above, can you implement this compositionally by calling `self.vec.into_boxed_slice` and then transmuting that to a `Box<str>`?